modules/shared/src/main/scala/org/tessellation/merkletree/MerkleTree.scala

```diff
@@ -28,9 +35,13 @@ case class Proof(entries: NonEmptyList[ProofEntry]) {
       .isDefined
 }
 
+@derive(show, encoder, decoder, eqv)
+case class MerkleRoot(leafCount: NonNegInt, hash: Hash)
```

Comment:
Is `leafCount` used anywhere?

Reply:
for Eq comparison

Reply:
Ok, but why is it necessary? Would some validation fail if we wouldn't store the leaf count? If so do we have a test case that can prove it?

Reply:
I have removed `leafCount` and all tests were passing but one test had missing expect on a merkle root. when I added it, the test is failing if we don't use `leafCount`.
![image](https://user-images.githubusercontent.com/4959931/236815760-05f13523-5935-4cdb-9a13-479b2a159f59.png)

merkle_tree("a", "b", "c") and merkle_tree("a", "b", "c", "c") produces the same root hash. we can either check the leaf count, or somehow force balanced trees only

Reply:
I've updated the test case

Reply:
Ok, thanks for the info